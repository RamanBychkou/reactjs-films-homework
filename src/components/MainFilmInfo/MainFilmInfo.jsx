import React from 'react';
import PropTypes from 'prop-types';
import FilmButtons from '../FilmButtons';
import style from './MainFilmInfo.scss';

const MainFilmInfo = (props) => {
  const { overview } = props;
  return (
    <div>
      <div className={style.mainFilmInfo__info} id="mainFilmOverwie">
        <p>{overview}</p>
      </div>
      <FilmButtons className={style.mainFilmInfo__buttons} />
    </div>
  );
};

MainFilmInfo.defaultProps = {
  overview: '',
};

MainFilmInfo.propTypes = {
  overview: PropTypes.string,
};

export default MainFilmInfo;
