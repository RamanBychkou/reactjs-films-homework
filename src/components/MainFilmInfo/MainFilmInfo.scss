.mainFilmInfo__info {
  color: #fff;
  display: flex;
  flex-flow: column nowrap;
}

.mainFilmInfo__buttons {
  color: #fff;
}

.mainFilmInfo__info p {
  text-overflow: ellipsis;
  max-height: 6.85rem;
  font-size: 0.73rem;
  letter-spacing: 0.2px;
  width: 13rem;
  align-self: flex-end;
  word-break: keep-all;
  line-height: 1.4rem;
  background-color: rgba(0, 0, 0, 0.64);
  padding: 0.7rem 0.5rem 0 0.99rem;
  overflow: hidden;
  font-weight: 700;
}
