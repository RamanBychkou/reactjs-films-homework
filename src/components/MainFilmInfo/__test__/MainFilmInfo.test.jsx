import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import ReactTestUtils, { act } from 'react-dom/test-utils';
import ShallowRenderer from 'react-test-renderer/shallow';
import configureStore from 'redux-mock-store';
import MainFilmInfo from '../index';

test('MainFilmInfo renders correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<MainFilmInfo />);
  const result = renderer.render(<MainFilmInfo />);
  expect(result).toMatchSnapshot();
});

test('MainFilmInfo: check state', () => {
  const props = {
    overwie: '',
    mainMovie: { id: 'test' },
  };

  const initialState = {
    movies: {
      page: 0,
      results: [],
      mainMovie: {},
      currentVideo: null,
    },
    mainMovie: { id: 'test' },
  };
  const mockStore = configureStore();
  const store = mockStore(initialState);
  const container = document.createElement('div');
  container.id = 'wrapper';
  document.body.appendChild(container);

  act(() => {
    ReactDOM.render(
      <Provider store={store}>
        <MainFilmInfo {...props} />
      </Provider>,
      document.querySelector('#wrapper'),
    );
  });
  const node = document.querySelector('#info');
  ReactTestUtils.Simulate.click(node);
  const result = document.querySelector('#wrapper');
  expect(result).toMatchSnapshot();
});
