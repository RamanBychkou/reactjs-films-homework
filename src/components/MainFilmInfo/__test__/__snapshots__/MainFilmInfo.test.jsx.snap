// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainFilmInfo renders correctly 1`] = `
<div
  className="hideInfo"
  id="mainFilmOverwie"
>
  <div
    className="mainFilmInfo__info"
  >
    <p>
      
    </p>
  </div>
  <Memo(ConnectFunction)
    className="mainFilmInfo__buttons"
    setTypeOverwie={[Function]}
  />
</div>
`;

exports[`MainFilmInfo: check state 1`] = `
<div
  id="wrapper"
>
  <div
    class="showInfo"
    id="mainFilmOverwie"
  >
    <div
      class="mainFilmInfo__info"
    >
      <p />
    </div>
    <div
      class="btnContainer"
    >
      <button
        class="watch"
        id="test"
        tabindex="0"
        type="button"
      >
        Watch Now
      </button>
      <button
        class="viewInfo"
        id="info"
        tabindex="0"
        type="button"
      >
        View Info
      </button>
    </div>
  </div>
</div>
`;
